### PR TITLE
Cert courses page

### PR DESCRIPTION
**Pull request by Havkost**
## Description

Create course list page and polish certain general components.

PBIs:
[Web Home] As a content creator, I want to view when I don't have an active course (empty state) and be directed to the creation environment
[Web Home] As a content creator, I want to view my created courses and their status
[Web Style] As content creator i would like to see consistent styling across different OS settings.

## Changes

- Made card view for viewing a user's courses
  - Added a sweet ass hover animation
  - Added StarRating component for viewing a rating as stars
  - Added LastEdited component for viewing last edited date/time
  - Added categories and a scalable icon system
  - Added colour coded statuses
- Added personal insights component with mock data
- Made empty state when a user has no courses
- Made navbar dark/light-mode compatible
  - Added Educado logos for each mode
  - Fixed text colour
- Added favicon(s)

## Related Issues

None.

## Checklist

- [x] Code has been tested locally and passes all relevant tests.
- [x] Documentation has been updated to reflect the changes, if applicable.
- [x] Code follows the established coding style and guidelines of the project.
- [x] All new and existing tests related to the changes have passed.
- [x] Any necessary dependencies or new packages have been properly documented.
- [x] Pull request title and description are clear and descriptive.
- [x] Reviewers have been assigned to the pull request.
- [x] Any potential security implications have been considered and addressed.
- [x] Performance impact of the changes has been evaluated, if relevant.

## Screenshots (if applicable)
image
educado-course-card-hover2

## Notes for Reviewers

This version was set up using mock data (which is now in a comment in pages/Courses.tsx) and a version of the backend circumventing JWT authentication. This needs to be implemented once the backend is updated to match the model in resources/MODELS.md.

We've added everyone from the teams we are sure have been working in this repo as reviewers. We only need 2, just write in discord if you're up for doing a review.

### CODE REVIEW

**Comment by Havkost** [comment]
We're aware that type checks are failing, we're on it 😊

File: app/src/services/course.services.ts [hunk 1]
```diff
@@ -1,59 +1,74 @@
...
   return await axios.post(
-    `${backend_route}/api/courses`,
+    `${BACKEND_URL}/api/courses`,
     {
       title: title,
       description: description,
-    },
-    { headers: { Authorization: `Bearer ${token}` } }
+    }/*,
+    { headers: { Authorization: `Bearer ${token}` } }*/
```
Comment (JackSMunn) [inline app/src/services/course.services.ts:27]: dead code or reimplement with JWT

File: app/src/services/course.services.ts [hunk 2]
```diff
@@ -1,59 +1,74 @@
...
+      });
+      return res.data;
+    });
 };
 
 // Get course detail
-const getCourseDetail = (url: string, token: string) => {
-  return axios.get(url, { headers: { Authorization: `Bearer ${token}` } })
+const getCourseDetail = (url: string/*, token: string*/) => {
+  return axios.get(url/*, { headers: { Authorization: `Bearer ${token}` } }*/)
```
Comment (JackSMunn) [inline app/src/services/course.services.ts:46]: reimplement JWT

File: app/src/services/course.services.ts [hunk 3]
```diff
@@ -1,59 +1,74 @@
...
     .then((res) => res.data);
 };
 
 // Get course categories
-const getCourseCategories = (url: string, token: string) => {
-  return axios.get(url, { headers: { Authorization: `Bearer ${token}` } })
+const getCourseCategories = (url: string/*, token: string*/) => {
+  return axios.get(url/*, { headers: { Authorization: `Bearer ${token}` } }*/)
```
Comment (JackSMunn) [inline app/src/services/course.services.ts:52]: Reimplement JWT

File: app/src/services/course.services.ts [hunk 4]
```diff
@@ -1,59 +1,74 @@
...
-  return axios.put(
-    `${import.meta.env.VITE_BACKEND_URL}api/courses/${id}`, // TODO: change backend url to not include final /
-    data,
-    { headers: { Authorization: `Bearer ${token}` } }
+const updateCourseDetail = (data: any, id: any/*, token: string*/) => {
+  console.log(`${BACKEND_URL}api/courses/update/${id}`)
+  return axios.post(
+    `${BACKEND_URL}/api/courses/update/${id}`, // TODO: change backend url to not include final /
+    data/*,
+    { headers: { Authorization: `Bearer ${token}` } }*/
```
Comment (JackSMunn) [inline app/src/services/course.services.ts:62]: Reimplement JWT

File: app/src/services/course.services.ts [hunk 5]
```diff
@@ -1,59 +1,74 @@
...
 
 // Create a new section for a course FIXME: should this be in section.services ??
 const createSection = async (data: any, id: any, token: string) => {
   return await axios.post(
-    `${backend_route}/api/courses/${id}/sections`,
-    data,
-    { headers: { Authorization: `Bearer ${token}` } }
+    `${BACKEND_URL}/api/sections/create/${id}`,
+    data/*,
+    { headers: { Authorization: `Bearer ${token}` }}*/
```
Comment (JackSMunn) [inline app/src/services/course.services.ts:71]: reimplement JWT

File: app/src/services/section.services.ts [hunk 6]
```diff
@@ -3,25 +3,26 @@ import axios from "axios";
...
     // Send the info to caller
-    return axios.put(
-        `${backend_url}/sections/${sid}`,
-        props,
-        { headers: { Authorization: `Bearer ${token}` } }
+    console.log(`${BACKEND_URL}/api/sections/update/${id}`);
+    return axios.post(
+        `${BACKEND_URL}/api/sections/update/${id}`,
+        data/*,
+    { headers: { Authorization: `Bearer ${token}` } }*/
```
Comment (JackSMunn) [inline app/src/services/section.services.ts:25]: JWT

**Review by JackSMunn — APPROVED** [review]
Reimplement JWT when merging, other than that, very beautiful mr. sea-broom

**Review by JackSMunn — COMMENTED** [review]
Re-checked code after newer commits. Still looks good.

**Comment by sofiiagran** [comment]
the code looks good ! however when running the website I am stuck on the login page, any way I can see the actual course screen on the web?

**Review by MaizeHH — APPROVED** [review]
Looks good, however some of the tabs/spacing is inconsistent (2, 4, 8 spaces).

**Comment by Jatewo** [comment]
> Looks good, however some of the tabs/spacing is inconsistent (2, 4, 8 spaces).

I think we should fix the linting before starting the next sprint. I'll bring this up with the integration team. So let it be for now

**Pull request merged**